program/rust/src/lib.rs

```diff
@@ -33,11 +36,34 @@ pub extern "C" fn c_entrypoint(input: *mut u8) -> u64 {
 
 #[no_mangle]
 pub extern "C" fn entrypoint(input: *mut u8) -> u64 {
-    let c_ret_val = unsafe { c_entrypoint(input) };
-    if c_ret_val == c_oracle_header::SUCCESSFULLY_UPDATED_AGGREGATE {
-        //0 is the SUCCESS value for solana
-        return 0;
-    } else {
-        return c_ret_val;
+    let (program_id, accounts, instruction_data) = unsafe { deserialize(input) };
+    let cmd_hdr_size = ::std::mem::size_of::<c_oracle_header::cmd_hdr>();
+    if instruction_data.len() < cmd_hdr_size {
+        panic!("insufficient data, could not parse instruction");
+    }
+
+    let cmd_data =
+        c_oracle_header::cmd_hdr::try_from_slice(&instruction_data[..cmd_hdr_size]).unwrap();
```

Comment:
I think here it's probably better to just do `use c_oracle_header::*` and have access to all the structs.

Reply:
I think it's good to have a clear visual seperation between C and Rust stuff. but happy to change it if you are sure.
use * still sounds not great to be me, but maybe use se c_oracle_header::{val1, val2 ..}